### ISSUE TITLE
DOI to freeze version and prevents subsequent edits

### ISSUE THREAD

**Issue by grosscol** [mlibrary/umrdr#52]
The final "publish" or whatever we're calling minting a DOI for a work should freeze it in place.  Since the repository is currently billed as existing for publishable work, **we can punt on making edits to a work after it has been given a DOI.**

The current issue is: Object with DOI has to be immutable.  Which should be a pretty easy check in a view.  The implementation that is accessible might be thornier.  @justcolin what do you think?

Subsequent issues will involve making changes to a Work that has a DOI will make a new version and remove the DOI for the new version.  This entanglement between DOI and version number has edge cases that should be addressed in subsequent tickets.

**Comment by grosscol** [comment]
Dependent on #9

**Comment by justcolin** [comment]
Correct me if I am wrong, but this means the view for the edit page has to check if the object has a DOI, and if it does, it has to make all fields uneditable and remove the submit button? Are any pages besides the edit page effected?

If so, that is easily doable.

**Comment by grosscol** [comment]
@justcolin that is spot on.  Since the DOI will be indexed, you should be able to check the presenter.

**Comment by grosscol** [comment]
Functionally, the immutable part of a doi'd work has to be the FileSets.  Specifically, the binary attached to them.  However, Jim pointed out that the metadata for the Work may need to change after the doi has been created.  For example, an author name might be misspelled. 

So subsequent edits to the FileSets and deleting should be disallowed at first pass.  The metadata for the Work can be left mutable.

**Comment by njaffer** [comment]
duplicate of #482
